### PR TITLE
TS-3874: Header-rewrite: support multiple header values in conditionals

### PR DESCRIPTION
**Pull request by jacksontj**

### CODE REVIEW

File: doc/reference/plugins/header_rewrite.en.rst [hunk 1]
```diff
@@ -208,3 +208,44 @@ Examples
   rm-header Set-Cookie
   counter plugin.header_rewrite.x-y-foobar-dc1
   cond %{HEADER:X-Y-Foobar} "Some string" [AND,NC]
+
+
+Notes about header conditionals
```
Comment (jpeach) [inline doc/reference/plugins/header_rewrite.en.rst]: Use the ```.. note::``` markup here.

File: doc/reference/plugins/header_rewrite.en.rst [hunk 2]
```diff
@@ -208,3 +208,44 @@ Examples
...
+-------------------------------
+In HTTP multple headers can be consolidated into a single comma separated string.
+To avoid complex markup within header-rewrite all conditionals against headers
```
Comment (jpeach) [inline doc/reference/plugins/header_rewrite.en.rst]: This sentence doesn't scan that well ... is there a word missing?

File: plugins/header_rewrite/conditions.cc [hunk 3]
```diff
@@ -237,13 +236,22 @@ ConditionHeader::append_value(std::string &s, const Resources &res)
...
     field_loc = TSMimeHdrFieldFind(bufp, hdr_loc, _qualifier.c_str(), _qualifier.size());
     TSDebug(PLUGIN_NAME, "Getting Header: %s, field_loc: %p", _qualifier.c_str(), field_loc);
-    if (field_loc != NULL) {
+
+    while (field_loc) {
       value = TSMimeHdrFieldValueStringGet(bufp, hdr_loc, field_loc, -1, &len);
+      next_field_loc = TSMimeHdrFieldNextDup(res.bufp, res.hdr_loc, field_loc);
       TSDebug(PLUGIN_NAME, "Appending HEADER(%s) to evaluation value -> %.*s", _qualifier.c_str(), len, value);
       s.append(value, len);
+      // multiple headers with the same name must be symantically the same as one value which is comma seperated (http://tools.ietf.org/html/rfc2616#section-4.2)
```
Comment (jpeach) [inline plugins/header_rewrite/conditions.cc]: *semantically*

**Comment by sudheerv** [comment]
lgtm

**Pull request merged**